File: src/main/java/chat/teco/tecochat/comment/domain/CommentRepository.java
package chat.teco.tecochat.comment.domain;

import static chat.teco.tecochat.comment.execption.CommentExceptionType.NOT_FOUND_COMMENT;

import chat.teco.tecochat.comment.execption.CommentException;
import java.util.List;
import org.springframework.data.jpa.repository.JpaRepository;

public interface CommentRepository extends JpaRepository<Comment, Long> {

    List<Comment> findAllByChatId(Long chatId);

    default Comment getById(Long id) {
        return findById(id)
                .orElseThrow(() -> new CommentException(NOT_FOUND_COMMENT));
    }
}

File: src/main/kotlin/chat/teco/tecochat/domain/comment/CommentRepository.kt
package chat.teco.tecochat.domain.comment

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.repository.findByIdOrNull

fun CommentRepository.getByIdOrThrow(id: Long) = findByIdOrNull(id)
    ?: throw NoSuchElementException("댓글이 존재하지 않습니다. id: $id")

interface CommentRepository : JpaRepository<Comment, Long> {

    fun findAllByChatId(chatId: Long): List<Comment>
}
